File: Programmers_42587/solution.js
function solution(priorities, location) {
  var answer = 0;
  const priorities_boolean = [];
  for (let i = 0; i < priorities.length; i++) {
    if (i === location) {
      priorities_boolean.push(true);
    } else {
      priorities_boolean.push(false);
    }
  }
  const printList = [];
  const boolList = [];
  while (priorities.length > 0) {
    const shift_item = priorities.shift();
    const shift_boolean = priorities_boolean.shift();
    if (priorities.length === 0) {
      printList.push(shift_item);
      boolList.push(shift_boolean);
      break;
    }
    for (let i = 0; i < priorities.length; i++) {
      if (priorities[i] > shift_item) {
        priorities.push(shift_item);
        priorities_boolean.push(shift_boolean);
        break;
      }
      if (i === priorities.length - 1) {
        printList.push(shift_item);
        boolList.push(shift_boolean);
      }
    }
  }

  answer = boolList.indexOf(true) + 1;

  return answer;
}

solution([1, 1, 9, 1, 1, 1], 0);

File: Programmers_42587/solution.py
def solution(priorities, location):
    # 동일한 우선 순위 구별 위해 idx부여
    priorities_with_idx = [(p, idx) for idx, p in enumerate(priorities)]
    print_list = []
    while priorities_with_idx :
        pop_item = priorities_with_idx.pop(0)
        for p in priorities_with_idx :
            if p[0] > pop_item[0] :
                priorities_with_idx.append(pop_item)
                pop_item = None

                break
                
        if pop_item != None :
            if pop_item[1] == location :
                return len(print_list)+1
            else :
                print_list.append(pop_item)